README.md

```diff
@@ -322,6 +323,35 @@ function Button(props) {
 }
 ```
 
+### Compose
+
+The `compose` API provides an optimisation that will compose your styles using reusable atomic **classes** and fallback to the `style` attribute when necessary for overrides.
```

Comment:
Are they really atomic though? 😄 I always thought that "atomic classes" means splitting rules into separate classes, here we're grouping all component classes together? 🤔 I probably know what you meant when looking into the `examples/remix`. The rules are kind-of reused and only variables values are set on classes, but this could be a little misleading

Reply:
they are atomic, the only reason they're grouped together is because lightningcss groups things that all share the same class or style declaration to reduce css size. the styles above originally looked something like the following before they were put through lightningcss:

```jsx
.tk4961385 {
  background-color: var(--background-color);
}

.tk4961385 {
  color: var(--color);
}

.tk3732cf61 {
  color: var(--color);
}

.tk4961385, .tk3732cf61 {
  font-family: var(--font-family);
}

.tk4961385 {
  --bg-color: var(--color_slate-700);
}

.tk4961385 {
  --color: var(--color_slate-100);
}

.tk3732cf61 {
  --color: var(--color_sky-500);
}

.tk4961385 {
  --font-family: var(--font_sans);
}

.tk3732cf61 {
  --font-family: var(--font_sans);
}
```

it looks like there might be a bug here tho:

![Screenshot 2024-01-15 at 12 06 17](https://github.com/tokenami/tokenami/assets/175330/a8a5928c-cf2d-493b-aa09-ffb6fcc4f05e)


i'd expect that to output what's below, so will looking into that before merging:

```css
.tk4961385 {
  background-color: var(--background-color);
}

.tk4961385, .tk3732cf61 {
  color: var(--color);
}
```

Reply:
Okay, I see your point, naming is hard 😄 I’m just wondering whether people will understand the tradeoff from the current phrasing

Reply:
can you elaborate what the tradeoff is that you see here? 

in tailwin, the css might have something like `.p-1 { padding: 1em; } .color-red { color: red } .color-blue { color: blue; }` and html like:

```html
<button class="p-1 color-red" />
<a class="p-1 color-blue" />
<!-- link button w/ twind-merge -->
<a class="p-1 color-red" />
```

 in tokenami it would be `.a, .b { padding: var(--padding); color: var(--color) }`, with html:

```html
<button class="a" />
<a class="b" />
<a class="a b" style="--color: red" />
```

Reply:
The stylesheet size, ability to cache it, and losing the hard limit on its size. Basically stuff that we discussed on Discord 😃